catalogd/internal/storage/localdir_test.go

```diff
@@ -13,219 +10,498 @@ import (
+		},
+		{
+			name:               "Server return 404 when path '/catalogs/<catalog>/non-existent.txt' is queried",
+			expectedStatusCode: http.StatusNotFound,
+			expectedContent:    "404 page not found",
+			URLPath:            "/catalogs/test-catalog/non-existent.txt",
+		},
+		{
+			name:               "Server returns 404 when path '/catalogs/<catalog>.jsonl' is queried even if the file exists, since we don't serve the filesystem, and serve an API instead",
+			expectedStatusCode: http.StatusNotFound,
+			expectedContent:    "404 page not found",
+			URLPath:            "/catalogs/test-catalog.jsonl",
+		},
+		{
+			name:               "Server returns 404 when non-existent catalog is queried",
+			expectedStatusCode: http.StatusNotFound,
+			expectedContent:    "404 Not Found",
+			URLPath:            "/catalogs/non-existent-catalog/api/v1/all",
+		},
+		{
+			name:               "Server returns 200 when path '/catalogs/<catalog>/api/v1/all' is queried, when catalog exists",
+			expectedStatusCode: http.StatusOK,
+			expectedContent:    `{"foo":"bar"}`,
+			URLPath:            "/catalogs/test-catalog/api/v1/all",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/%s", testServer.URL, tc.URLPath), nil)
+			require.NoError(t, err)
+			req.Header.Set("Accept-Encoding", "gzip")
+			resp, err := http.DefaultClient.Do(req)
+			require.NoError(t, err)
+
+			require.Equal(t, tc.expectedStatusCode, resp.StatusCode)
+
+			var actualContent []byte
+			actualContent, err = io.ReadAll(resp.Body)
+			require.NoError(t, err)
+			require.Equal(t, tc.expectedContent, strings.TrimSpace(string(actualContent)))
+			require.NoError(t, resp.Body.Close())
 		})
-		When("The stored content is deleted", func() {
-			BeforeEach(func() {
-				err := store.Delete(catalog)
-				Expect(err).To(Not(HaveOccurred()))
-			})
-			It("should delete the FBC from the cache directory", func() {
-				fbcFile := filepath.Join(rootDir, catalog)
-				_, err := os.Stat(fbcFile)
-				Expect(err).To(HaveOccurred())
-				Expect(os.IsNotExist(err)).To(BeTrue())
-			})
-			It("should report content does not exist", func() {
-				Expect(store.ContentExists(catalog)).To(BeFalse())
-			})
+	}
+}
+
+// Tests to verify the behavior of the query endpoint, as described in
+// https://docs.google.com/document/d/1s6_9IFEKGQLNh3ueH7SF4Yrx4PW9NSiNFqFIJx0pU-8/edit?usp=sharing
+func TestQueryEndpoint(t *testing.T) {
+	store := &LocalDirV1{
+		RootDir:            t.TempDir(),
+		RootURL:            &url.URL{Path: urlPrefix},
+		EnableQueryHandler: true,
+	}
+	if store.Store(context.Background(), "test-catalog", createTestFS(t)) != nil {
+		t.Fatal("failed to store test catalog")
+	}
+	testServer := httptest.NewServer(store.StorageServerHandler())
+
+	testCases := []struct {
+		name               string
+		setupStore         func() (*httptest.Server, error)
+		queryParams        string
```

Comment:
Nit: might be nicer to use `url.Values` for more structure setup of the query params.